lib/rubocop/cop/factory_bot/factory_association_with_strategy.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module FactoryBot
+      # Use definition in factory association instead of hard coding a strategy.
+      #
+      # @example
+      #   # bad - only works for one strategy
+      #   factory :foo do
+      #     profile { create(:profile) }
+      #   end
+      #
+      #   # good - implicit
+      #   factory :foo do
+      #     profile
+      #   end
+      #
+      #   # good - explicit
+      #   factory :foo do
+      #     association :profile
+      #   end
+      #
+      #   # good - inline
+      #   factory :foo do
+      #     profile { association :profile }
+      #   end
+      #
+      class FactoryAssociationWithStrategy < ::RuboCop::Cop::Base
+        FACTORY_CALLS = RuboCop::FactoryBot::Language::METHODS
+
+        MSG = 'Use an implicit, explicit or inline definition instead of ' \
+              'hard coding a strategy for setting association within factory.'
+
+        # @!method factory_strategy_association(node)
+        def_node_matcher :factory_strategy_association, <<~PATTERN
+          (block (send nil? :factory ...)
+            ...
+          )
+        PATTERN
+
+        def on_block(node)
+          factory_strategy_association(node) do
```

Comment:
Then, you could use the matched `create(...)` here directly:
```suggestion
          factory_strategy_association(node) do |node|
```